src/internals/module_import.rs

```diff
@@ -0,0 +1,45 @@
+use std::ffi::{c_void, CStr};
+use std::fmt::Debug;
+
+/// Details about a module being imported.
+pub struct YrModuleImport<'a>(&'a mut yara_sys::YR_MODULE_IMPORT);
+
+impl Debug for YrModuleImport<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        Debug::fmt("YrModuleImport", f)
+    }
+}
+
+impl<'a> From<&'a mut yara_sys::YR_MODULE_IMPORT> for YrModuleImport<'a> {
+    fn from(value: &'a mut yara_sys::YR_MODULE_IMPORT) -> Self {
+        Self(value)
+    }
+}
+
+impl YrModuleImport<'_> {
+    /// Get the name of the module.
+    pub fn name(&self) -> Option<&[u8]> {
+        let ptr = self.0.module_name;
+        if ptr.is_null() {
+            None
+        } else {
+            // Safety:
+            // - ptr is not null, and is guaranteed by libyara to be nul-terminated
+            // - returned slice is valid for as long as self, guaranteeing the ptr to stay valid.
+            let cstr = unsafe { CStr::from_ptr(ptr) };
+            Some(cstr.to_bytes())
+        }
+    }
+
+    /// Set the module data to be used by the module.
+    ///
+    /// # Safety
+    ///
+    /// The caller must guarantee that:
+    /// - `ptr` is valid for reads of `size` bytes.
+    /// - `ptr` stays valid for the full duration of the scan.
+    pub unsafe fn set_module_data(&mut self, ptr: *mut c_void, size: u64) {
```

Comment:
@vthib hello! Are you sure to make a raw pointer as a public interface?

Reply:
I don't really see any other way, making this function safe would mean changing this pointer to a borrowed ref, but it would need to be tied to a lifetime that outlives the borrow on `Rules`/`Scanner`, which i don't think would be easy to express here. And even then, there isn't really a type to expose, since the type depends on the module.

I feel like exposing the raw pointer, with a clear unsafe function, is the best trade-off here. This is a very niche use-case, and leaving it to the user to properly set it makes the API changes minimal and still make it possible to set this module data if needed